x-pack/legacy/plugins/reporting/server/lib/validate/validate_server_host.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ServerFacade } from '../../../types';
+
+const configKey = 'xpack.reporting.kibanaServer.hostname';
+
+export function validateServerHost(serverFacade: ServerFacade) {
+  const config = serverFacade.config();
+
+  const serverHost = config.get('server.host');
+  const reportingKibanaHostName = config.get(configKey);
+
+  if (!reportingKibanaHostName && serverHost === '0') {
+    // @ts-ignore: No set() method on KibanaConfig, just get() and has()
```

Comment:
This is brought up as a concern in the Reporting migration issue: https://github.com/elastic/kibana/issues/53898